go/blindtree/values.go

```diff
@@ -0,0 +1,78 @@
+package blindtree
+
+import (
+	"github.com/keybase/client/go/protocol/keybase1"
+	"github.com/keybase/client/go/sig3"
+	"github.com/keybase/go-codec/codec"
+)
+
+// BlindMerkleValue simulates a union type to store values of different types in
+// the Blind Merkle Tree. To add a new type, grab a new BlindMerkleValueType
+// constant and update the CodecEncodeSelf and CodecDecodeSelf functions as
+// appropriate.
+type BlindMerkleValue struct {
+	ValueType  BlindMerkleValueType
+	InnerValue interface{}
+}
+
+// Note: values up to 127 are preferred as they are encoded in a single byte
+type BlindMerkleValueType uint8
+
+const (
+	ValueTypeEmpty  BlindMerkleValueType = 0
+	ValueTypeTeamV1 BlindMerkleValueType = 1
+
+	ValueTypeStringForTesting BlindMerkleValueType = 127
+)
+
+var _ codec.Selfer = &BlindMerkleValue{}
+
+func (t *BlindMerkleValue) CodecEncodeSelf(e *codec.Encoder) {
+	switch t.ValueType {
+	case ValueTypeTeamV1, ValueTypeStringForTesting:
+		// pass
+	default:
+		panic("Unknown merkle value type")
+	}
+	e.MustEncode(t.ValueType)
+	e.MustEncode(t.InnerValue)
+}
+
+func (t *BlindMerkleValue) CodecDecodeSelf(d *codec.Decoder) {
+	d.MustDecode(&t.ValueType)
+	switch t.ValueType {
+	case ValueTypeEmpty:
+		// Nothing to do here
+	case ValueTypeTeamV1:
+		var v Teamv1Value
+		d.MustDecode(&v)
+		t.InnerValue = v
+	case ValueTypeStringForTesting:
+		var s string
+		d.MustDecode(&s)
+		t.InnerValue = s
+	default:
+		panic("Unrecognized Value Type")
+	}
+}
+
+type SigID []byte
+type Teamv1HiddenTail struct {
+	_struct struct{} `codec:",toarray"` //nolint
+	SigID   SigID
+	LinkID  sig3.LinkID
+	Seqno   keybase1.Seqno
```

Comment:
What was the verdict about including SigID here? Do we need it?

Reply:
Removing both SigID and Seqno as we discussed. cc @maxtaco